test/splunk/test_prepare.py

```diff
@@ -73,6 +73,62 @@ def test_prepare_trap(self, m_custom):
         self.assertEqual(67.0, event2["metric_three"]["value"])
         self.assertEqual(90.0, event2["metric_four"]["value"])
 
+    @patch("splunk_connect_for_snmp.splunk.tasks.SPLUNK_AGGREGATE_TRAPS_EVENTS", True)
+    @patch("splunk_connect_for_snmp.splunk.tasks.apply_custom_translations")
+    def test_prepare_aggregated_trap(self, m_custom):
+        task_input = {
+            "sourcetype": "sc4snmp:traps",
+            "time": 1234567,
+            "address": "192.168.0.1",
+            "result": {
+                "SOME_GROUP_KEY1": {
+                    "metrics": {
+                        "metric_one": {"value": 23},
+                        "metric_two": {"value": 26},
+                    },
+                    "fields": {
+                        "field_one": {"value": "on"},
+                        "field_two": {"value": "listening"},
+                    },
+                },
+                "SOME_GROUP_KEY2": {
+                    "metrics": {
+                        "metric_three": {"value": 67},
+                        "metric_four": {"value": 90},
+                    },
+                    "fields": {
+                        "field_three": {"value": "OFF"},
+                        "field_four": {"value": "stopping"},
+                    },
+                },
+            },
+        }
+
+        m_custom.return_value = task_input
+        result = prepare(task_input)
+
+        self.assertEqual("list", type(result["events"]).__name__)
+        self.assertEqual(2, len(result))
+        self.assertEqual(1, len(result["events"]))
+        self.assertEqual(0, len(result["metrics"]))
+
+        item1 = json.loads(result["events"][0])
+        event1 = json.loads(item1["event"])
```

Comment:
Can we change it into a real json object so it's easy to see how the final structure looks like?

Reply:
done